pkg/chaosdaemon/server.go

```diff
@@ -45,22 +45,26 @@ func newServer() (*Server, error) {
 	}, nil
 }
 
-func StartServer(host string, port int) {
+// StartServer starts chaos-daemon.
+func StartServer(host string, port int, containerRuntime string) {
 	lis, err := net.Listen("tcp", fmt.Sprintf("%s:%d", host, port))
 	if err != nil {
 		log.Error(err, "failed to listen")
+		return
 	}
 
 	s := grpc.NewServer()
-	chaosDaemonServer, err := newServer()
+	chaosDaemonServer, err := newServer(containerRuntime)
 	if err != nil {
 		log.Error(err, "failed to create server")
+		return
 	}
 	pb.RegisterChaosDaemonServer(s, chaosDaemonServer)
 
 	reflection.Register(s)
 
 	if err := s.Serve(lis); err != nil {
 		log.Error(err, "failed to serve")
+		return
```

Comment:
Can I ask why we don't add return when there is an error in this function? Is it a bug?

Reply:
@YangKeao @cwen0 😄

Reply:
@yeya24 Because of The Spring Festival, maybe we did not response very quickly.

Reply:
It's strange don't return when meet error. But if we don't return anything just stop this function, the process will exit normally. What about return an error in this function?

Reply:
I think here we should add return, I think it is a bug.

Reply:
Yes, this is a bug, I think we should return the error and handle this error in main function.